sql/variables/status_variables.go

```diff
@@ -0,0 +1,4181 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package variables
+
+import (
+	"sync"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// globalStatusVariables is the underlying type of SystemVariables.
+type globalStatusVariables struct {
+	mutex   *sync.RWMutex
+	varVals map[string]sql.StatusVarValue
+}
+
+var _ sql.StatusVariableRegistry = (*globalStatusVariables)(nil)
+
+// NewSessionMap implements sql.StatusVariableRegistry
+func (g *globalStatusVariables) NewSessionMap() map[string]sql.StatusVarValue {
```

Comment:
Related to the comment in `status_variable_test.go` about keeping session and global vars accessed separately... it seems like `NewSessionMap` should probably only copy over status variables that have a session scope (or have both session and global scope). That seems like it'll be a cleaner calling pattern if global status vars are always accessed through a global interface and session status vars are always accessed through the session, plus it means we have to copy over less status vars for each session, which is nice.